### PR TITLE
convert from using uint32 to uint16 in array and run containers

### LINKED ISSUE #664: Use 16 bit ints instead of 32 in array and rle containers
### Description
We currently use uint32 for array and rle containers in the roaring implementation, despite all values fitting in to uint16. Using uint16 should reduce the amount of storage used, and may boost performance.

### Success criteria (What criteria will consider this ticket closeable?)
uint16 are used instead of uint32 and storage is reduced.

### PR DESCRIPTION
**Pull request by jaffee**
## Overview

See title and issue - wait on this until RLE is merged.

Fixes #664

## Pull request checklist

- [ ] I have read the [contributing guide](https://github.com/pilosa/pilosa/blob/master/CONTRIBUTING.md).
- [ ] I have agreed to the [Contributor License Agreement](https://cla-assistant.io/pilosa/pilosa).
- [ ] I have updated the [documentation](https://github.com/pilosa/pilosa/tree/master/docs).
- [ ] I have resolved any merge conflicts.
- [ ] I have included tests that cover my changes.
- [ ] All new and existing tests pass.

## Code review checklist
This is the checklist that the reviewer will follow while reviewing your pull request. You do not need to do anything with this checklist, but be aware of what the reviewer will be looking for.

- [ ] Ensure that any changes to external docs have been included in this pull request.
- [ ] If the changes require that minor/major versions need to be updated, tag the PR appropriately.
- [ ] Ensure the new code is [properly commented](https://github.com/golang/go/wiki/CodeReviewComments#doc-comments) and follows [Idiomatic Go](https://dmitri.shuralyov.com/idiomatic-go).
- [ ] Check that tests have been written and that they cover the new functionality.
- [ ] Run tests and ensure they pass.
- [ ] Build and run the code, performing any applicable integration testing.

### CODE REVIEW

**Comment by jaffee** [comment]
@travisturner On that line, we're trying to decide if it would be more space efficient to convert to an array container to a run container. Each run takes up twice the space of an element of an array, so if the number of runs is less than half the array cardinality, then using RLE is more efficient.

TLDR; I think `2` is correct there

**Comment by travisturner** [comment]
@jaffee but doesn't the change to 16-bit array values mean that each run takes up 4x the space of an array element?

**Comment by jaffee** [comment]
we changed to `interval16` for run containers as well

**Comment by travisturner** [comment]
oh, got it. sorry.

**Comment by tgruben** [comment]
closed due to #758
